core-v1/src/main/scala/data/ExtractValue.scala

```diff
@@ -15,9 +15,17 @@ trait ExtractValue[T, V] extends (T => V) {
 object ExtractValue {
   type AsBoolean[T] = ExtractValue[T, Boolean]
 
+  type As[V] = {
+    type From[T] = ExtractValue[T, V]
+  }
+
+  type From[T] = {
+    type As[V] = ExtractValue[T, V]
+  }
```

Comment:
Are these being used? If so, I should document them and maybe refactor old code to use them where appropriate. I suppose one advantage of these is that you don't need the kind-projector compiler plugin to add these as context bounds. However, long-term, I think these will be phased out in favor of kind-projector... so why add them now? Are they really used enough to make it worth adding them here?